tsl/src/continuous_aggs/materialize.c

```diff
@@ -459,7 +468,35 @@ get_materialization_end_point_for_table(int32 raw_hypertable_id, int32 materiali
 		*materializing_new_range = false;
 		return old_completed_threshold;
 	}
-
+	if (max_time < end_time)
+	{
+		int64 maxarg = max_time;
+		/* max_time needs to be adjusted by negative refresh_lag to put it
+		 * into the correct time_bucket. Say we have max_time = 100
+		 * and refresh_lag = -100, time_bucket width = 100
+		 * time_bucket(100, max_time) = 100
+		 * but time_bucket(100, max_time-refresh_lag) = 200
+		 */
+		if (refresh_lag < 0 && !(max_time >= PG_INT64_MAX + refresh_lag))
+			maxarg = max_time - refresh_lag;
+		/* overflow check */
+		Assert(bucket_width > 0);
+		if (!((maxarg <= PG_INT64_MIN + bucket_width) || (maxarg >= PG_INT64_MAX - bucket_width)))
```

Comment:
Can you add a comment explaining the condition to enter this if clause?

And, isn't this the same as:

```
if ((maxarg > PG_INT64_MIN + bucket_width) && (maxarg < PG_INT64_MAX - bucket_width))
```

I find this easier to understand that the inverted clause.